### PR TITLE
Migrate from dep to Go modules

### PR DESCRIPTION
**Pull request by tomahawk28**
## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module github.com/giantswarm/k8scloudconfig
+
+go 1.13
+
+require (
+	github.com/Masterminds/semver v1.5.0
+	github.com/ghodss/yaml v1.0.1-0.20190212211648-25d852aebe32 // indirect
+	github.com/giantswarm/apiextensions v0.1.0
```
Comment (xh3b4sd) [inline go.mod]: Our deps need to be on at least `v0.2.0`.

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	github.com/giantswarm/microerror v0.1.0
+	github.com/gogo/protobuf v1.3.1 // indirect
+	github.com/juju/errgo v0.0.0-20140925100237-08cceb5d0b53 // indirect
+	github.com/stretchr/testify v1.4.0
+	golang.org/x/net v0.0.0-20191207000613-e7e4b65ae663 // indirect
+	gopkg.in/inf.v0 v0.9.1 // indirect
+	gopkg.in/yaml.v2 v2.2.8
+	k8s.io/apiextensions-apiserver v0.0.0-20191114105449-027877536833 // indirect
```
Comment (xh3b4sd) [inline go.mod]: The k8s deps need to be at least on `v0.16.6`.

Comment (tomahawk28) [reply go.mod]: After updating all to 0.2.0 for gs libs, it's gone.

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+go 1.13
+
+require (
+	github.com/Masterminds/semver v1.5.0
+	github.com/giantswarm/apiextensions v0.2.0
+	github.com/giantswarm/microerror v0.2.0
+	github.com/gogo/protobuf v1.3.1 // indirect
+	github.com/stretchr/testify v1.4.0
+	golang.org/x/net v0.0.0-20191207000613-e7e4b65ae663 // indirect
```
Comment (xh3b4sd) [inline go.mod]: FYI you can usually remove these indirects and run `go mod tidy` again. It is usually fine. I encountered it sometimes adds some of them back and then it stays but that way we can try to keep it a bit more tidy.

**Comment by wtty-fool** [comment]
I think this needs the following treatment:
```
devctl replace -i --ignore './vendor/**' 'microerror\.Stack' 'microerror.JSON' '**/*.go'                                                                                              
devctl replace -i --ignore './vendor/**' 'microerror\.Maskf\(err,.*(\)\n)' 'microerror.Mask(err$1)' '**/*.go'
```

**Pull request merged**